apps/s_server.c

```diff
@@ -3330,6 +3348,43 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
                              "HTTP/1.0 200 ok\r\nContent-type: text/plain\r\n\r\n");
             }
             /* send the file */
+#ifndef OPENSSL_NO_KTLS
+            if (use_sendfile) {
+                FILE *fp = NULL;
+                int fd;
+                struct stat st;
+                off_t offset = 0;
+                size_t sz;
+
+                BIO_get_fp(file, &fp);
+                fd = fileno(fp);
+                if (fstat(fd, &st) < 0) {
```

Comment:
Should this `fstat` be covered by macro `#ifndef OPENSSL_NO_POSIX_IO`?

Reply:
And all other places related to `fstat` stuffs as well...

Reply:
> 
> 
> Should this `fstat` be covered by macro `#ifndef OPENSSL_NO_POSIX_IO`?

My opinion is that this is wrapped by #ifndef OPENSSL_NO_KTLS. Currently, only the operating systems of Linux and BSD systems implement KTLS, which is equivalent to including POSIX, so there is no need to add any more here.